File: 1904/A - Forked!/solution.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main() {
    int t;
    cin >> t;
 
    while (t--) {
        int a, b;
        cin >> a >> b;
 
        int xk, yk;
        cin >> xk >> yk;
 
        int xq, yq;
        cin >> xq >> yq;
 
        vector<pair<int, int>> moves = {
            {a, b}, {a, -b}, {-a, b}, {-a, -b},
            {b, a}, {b, -a}, {-b, a}, {-b, -a}
        };
 
        if (a == b) {
            moves = {
                {a, b}, {a, -b},
                {-a, b}, {-a, -b}
            };
        }
 
        int count = 0;
 
        for (auto [dx, dy] : moves) {
            int x = xk + dx;
            int y = yk + dy;
 
            if ((abs(x - xq) == a && abs(y - yq) == b) ||
                (abs(x - xq) == b && abs(y - yq) == a)) {
                count++;
            }
        }
 
        cout << count << '
';
    }
 
    return 0;
}

File: 1904/A - Forked!/solution.java
import java.util.Scanner;
public class Problem_1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        for (int test = 1; test <= t; test++) {
            int a = sc.nextInt();
            int b = sc.nextInt();
 
            int xk = sc.nextInt();
            int yk = sc.nextInt();
 
            int xq = sc.nextInt();
            int yq = sc.nextInt();
 
            int[][] moves = {
                    {a, b}, {a, -b}, {-a, b}, {-a, -b},
                    {b, a}, {b, -a}, {-b, a}, {-b, -a}
            };
 
            if (a == b) {
                int[][] newMoves = {
                        {a, b}, {a, -b}, {-a, b}, {-a, -b}
                };
                moves = newMoves;
            }
 
            int count = 0;
 
            for (int[] move : moves) {
                int x = xk + move[0];
                int y = yk + move[1];
 
                if ((Math.abs(x - xq) == a && Math.abs(y - yq) == b)
                        || (Math.abs(x - xq) == b && Math.abs(y - yq) == a)) {
                   count++;
                }
            }
            System.out.println(count);
        }
    }
}